src/cursor/abstract_cursor.ts

```diff
@@ -457,13 +458,22 @@ export abstract class AbstractCursor<
    * cursor.rewind() can be used to reset the cursor.
    */
   async toArray(): Promise<TSchema[]> {
-    const array = [];
+    const array: TSchema[] = [];
+    // at the end of the loop (since readBufferedDocuments is called) the buffer will be empty
+    // then, the 'await of' syntax will run a getMore call
     for await (const document of this) {
       array.push(document);
+      const docs = this.readBufferedDocuments();
+      if (this.transform != null) {
+        for (const doc of docs) {
+          array.push(await this.transformDocument(doc));
```

Comment:
The PR description says we no longer await each document when building the array.  That's true if there isn't a transform, but are we planning on also fixing the unnecessary `awaits` when there is a transform?

Reply:
I think the reply here: https://github.com/mongodb/node-mongodb-native/pull/4171#discussion_r1698991071 alludes to it not making a difference. @aditi-khare-mongoDB is that correct?

Maybe we should call out that the optimization only applies to cursors without a transform? 

I think the only way around this would either be making transform sync again or migrating logic from transform into toArray, which we can file a follow-up for, this still successfully addresses common usage.